grace/server.go

```diff
@@ -240,6 +248,7 @@ func (srv *Server) serverTimeout(d time.Duration) {
 			break
 		}
 		srv.wg.Done()
+		runtime.Gosched()
```

Comment:
which race condition? I want to know the root cause here while not use Gosched to ignore the race condition